python/cudf/cudf/core/reshape.py

```diff
@@ -449,11 +467,16 @@ def length_check(obj, name):
     else:
         result_df = df.drop(labels=columns)
         for name in columns:
-            if hasattr(df[name]._column, "categories"):
+            if isinstance(df[name]._column, CategoricalColumn):
                 unique = df[name]._column.categories
             else:
                 unique = df[name].unique()
 
+            if dummy_na is False:
+                if not isinstance(unique, StringColumn):
```

Comment:
Won't line 473 return a `Series`, so should we use `is_string_dtype()` instead here?

Reply:
Nice catch